src/rust/rust-proto-new/src/lib.rs

```diff
@@ -214,4 +220,70 @@ pub trait SerDe: type_url::TypeUrl + Clone + std::fmt::Debug {
         Self: Sized;
 }
 
-pub(crate) mod serde_impl;
+pub(crate) mod serde_impl {
+    use bytes::{
+        Bytes,
+        BytesMut,
+    };
+    use prost::Message;
+
+    use crate::{
+        type_url,
+        SerDe,
+        SerDeError,
+    };
+
+    /// You can use this blanket implementation to automatically implement SerDe
+    /// for a type YourType defined within this crate. YourType must implement
+    /// either From<proto::YourType> or TryFrom<proto::YourType> and you must
+    /// implement either From<YourType> or TryFrom<YourType> for
+    /// proto::YourType. YourType must also implement type_url::TypeUrl. If
+    /// YourType meets these conditions you can automatically implement SerDe by
+    /// implementing the simpler trait ProtobufSerializable.
+    ///
+    /// Example usage:
+    ///
+    /// use crate::{
+    ///     protobufs::graplinc::v1beta1 as proto,
+    ///     serde_impl,
+    ///     type_url,
+    /// };
+    ///
+    /// pub struct YourType {...}
+    ///
+    /// impl type_url::TypeUrl for YourType {
+    ///     const TYPE_URL: &'static str =
+    ///         "graplsecurity.com/graplinc.v1beta1.YourType";
+    /// }
+    ///
+    /// impl serde_impl::ProtobufSerializable<YourType> for YourType {
+    ///    type ProtobufMessage = proto::YourType;
+    /// }
+    pub(crate) trait ProtobufSerializable<T> {
+        type ProtobufMessage: TryFrom<T> + TryInto<T> + Message + Default;
+    }
+
+    impl<T> SerDe for T
+    where
+        T: ProtobufSerializable<T>,
+        T: type_url::TypeUrl + Clone + std::fmt::Debug,
+        SerDeError: From<<<T as ProtobufSerializable<T>>::ProtobufMessage as TryFrom<T>>::Error>,
+        SerDeError: From<<<T as ProtobufSerializable<T>>::ProtobufMessage as TryInto<T>>::Error>,
```

Comment:
I wish I was better at rust, i feel like there's a better way to do this, but it's fine I guess lol